includes/blocks/class-sensei-lesson-blocks.php

```diff
@@ -122,6 +122,8 @@ public function initialize_blocks() {
 		 */
 		$post_type_object->template = apply_filters( 'sensei_lesson_block_template', $block_template, $post_type_object->template ?? [] );
 
+		new Sensei_Conditional_Content_Block();
```

Comment:
Hmm, just asking for clarity, but shouldn't this block be after the checking `Sensei()->lesson->has_sensei_blocks()` just below?

Reply:
Ah, I think it's because the Conditional Content Block is not part of the check: `has_sensei_blocks`. So we could have that and continue with the `has_sensei_blocks` being `false`.

Reply:
Ahhhhh, fair enough!

While searching for `has_sensei_blocks()`, I found another method that only checked if the post content had a...well, sensei block.

Now while looking through my IDE, I found the correct method, and it indeed makes sense!

I'll just resolve the conversation then. :+1:

Reply:
That one only checks for the lesson actions or lesson properties blocks, so because of the early return this block wouldn't work if those were not present. We could add this block to the `has_sensei_blocks` check, but that functions is aimed at deciding if the lesson should be using the old template hooks for rendering the actions & co, or if it's a new block-based lesson. Those blocks provide the functionality that would be done by the hooks (eg. rendering the buttons). 

And we wouldn't want the lesson buttons to disappear when someone adds the conditional content block, as it's not one replacing template hooks.

Note that registering all those blocks after the `if` is just a tiny optimization, we could also move those up, they don't have any side-effect, it's just the `remove_block_related_content` that does the template hook un-hooking.